### PR TITLE
gh-103092: Isolate msvcrt

### PR DESCRIPTION
**Pull request by aisk**
* Issue: gh-103092

### CODE REVIEW

File: PC/msvcrtmodule.c [hunk 1]
```diff
@@ -664,10 +648,34 @@ PyInit_msvcrt(void)
...
+static struct PyModuleDef msvcrtmodule = {
+    PyModuleDef_HEAD_INIT,
+    "msvcrt",
+    NULL,
+    0,
+    msvcrt_functions,
+    msvcrt_slots,
+    NULL,
+    NULL,
+    NULL
```
Comment (arhadthedev) [inline PC/msvcrtmodule.c]: A useful feature of C99:

```suggestion
    .m_base = PyModuleDef_HEAD_INIT,
    .m_name = "msvcrt",
    .m_methods = msvcrt_functions,
    .m_slots = msvcrt_slots
```

File: Misc/NEWS.d/next/Library/2023-04-04-21-44-25.gh-issue-103092.Dz0_Xn.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Adapt :mod:`!msvcrt` to :pep:`687`.
```
Comment (erlend-aasland) [inline Misc/NEWS.d/next/Library/2023-04-04-21-44-25.gh-issue-103092.Dz0_Xn.rst]:
```suggestion
Adapt the :mod:`msvcrt` extension module to :pep:`687`.
```

File: PC/msvcrtmodule.c [hunk 3]
```diff
@@ -664,10 +648,29 @@ PyInit_msvcrt(void)
...
+static PyModuleDef_Slot msvcrt_slots[] = {
+    {Py_mod_exec, exec_module},
+    {0, NULL}
+};
+
+static struct PyModuleDef msvcrtmodule = {
+    .m_base = PyModuleDef_HEAD_INIT,
+    .m_name = "msvcrt",
+    .m_methods = msvcrt_functions,
+    .m_slots = msvcrt_slots
```
Comment (erlend-aasland) [inline PC/msvcrtmodule.c]: I prefer a trailing comma for all struct initialisers that could possibly be expanded in the future.

```suggestion
    .m_slots = msvcrt_slots,
```

File: PC/msvcrtmodule.c [hunk 4]
```diff
@@ -664,10 +648,29 @@ PyInit_msvcrt(void)
                                                   _VC_CRT_BUILD_VERSION,
                                                   _VC_CRT_RBUILD_VERSION);
     st = PyModule_AddObject(m, "CRT_ASSEMBLY_VERSION", version);
-    if (st < 0) return NULL;
+    if (st < 0) return -1;
```
Comment (erlend-aasland) [inline PC/msvcrtmodule.c]: We can apply PEP-7 to the code we touch; it is nice to leave the code in a better state than what we found it.

```suggestion
    if (st < 0) {
        return -1;
    }
```

Comment (erlend-aasland) [reply PC/msvcrtmodule.c]: You also need to change the three `return NULL`s above.

**Comment by erlend-aasland** [comment]
I'd like to land gh-103383 first.

File: PC/msvcrtmodule.c (hunk 4 above)
Comment (aisk) [reply PC/msvcrtmodule.c]: Thanks for review, their macro blocks were folded by the IDE so I missed them.

Comment (erlend-aasland) [reply PC/msvcrtmodule.c]: Yeah, I figured that :)

**Comment by erlend-aasland** [comment]
(Sorry, I messed up the merge!)

**Review by erlend-aasland — APPROVED** [review]

**Comment by erlend-aasland** [comment]
Thanks, @aisk.

**Pull request merged**